webui/src/lib/components/nav.jsx

```diff
@@ -1,40 +1,33 @@
 import React from "react";
-import {Link as RouterLink} from "react-router-dom";
+import {NavLink as RouterLink, useHref, useLinkClickHandler} from "react-router-dom";
 
 import Nav from "react-bootstrap/Nav";
 
 import {buildURL} from "../hooks/router";
 import {Tabs} from "@mui/material";
 
-
-function isModifiedEvent(event) {
-    return !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey);
-}
-
 const wrapComponent = (component) => {
-    const linkWrapper = React.forwardRef(({navigate, onClick, ...rest}, forwardedRef) => {
-        const {target} = rest;
+    const linkWrapper = React.forwardRef(({navigate, onClick, to, target, replace, state, ...rest}, ref) => {
+        const href = useHref(to);
+        const handleClick = useLinkClickHandler(to, {
+            replace,
+            state,
+            target,
+        });
+
         const props = {
             ...rest,
-            ref: forwardedRef,
-            onClick: event => {
-                try {
-                    if (onClick) onClick(event);
-                } catch (ex) {
-                    event.preventDefault();
-                    throw ex;
+            ref,
+            href,
+            onClick: (event) => {
+                onClick?.(event);
```

Comment:
Wanted to verify that we are ok with this one https://caniuse.com/mdn-javascript_operators_optional_chaining

Reply:
Well, I think it's equivalent to the previous implementation, except it doesn't throw an error, which I believe was pointless in the original implementation anyway, since no one handles it in any meaningful way.